src/components/NewCommentForm.tsx

```diff
@@ -1,69 +1,129 @@
-import React from 'react';
+import { useState } from 'react';
+import classNames from 'classnames';
+import { Input } from './Input';
+import { CommentData } from '../types/Comment';
 
-export const NewCommentForm: React.FC = () => {
-  return (
-    <form data-cy="NewCommentForm">
-      <div className="field" data-cy="NameField">
-        <label className="label" htmlFor="comment-author-name">
-          Author Name
-        </label>
+type FormData = {
+  name: string;
+  email: string;
+  body: string;
+};
 
-        <div className="control has-icons-left has-icons-right">
-          <input
-            type="text"
-            name="name"
-            id="comment-author-name"
-            placeholder="Name Surname"
-            className="input is-danger"
-          />
+type FormErrors = {
+  name: boolean;
+  email: boolean;
+  body: boolean;
+};
 
-          <span className="icon is-small is-left">
-            <i className="fas fa-user" />
-          </span>
+type NewCommentFormProps = {
+  postId?: number;
+  handleAddComment: (data: any, url: string, msg: string) => Promise<void>;
```

Comment:
Consider changing type of 'data' to 'CommentData' instead of 'any' for better type safety.